tests/DataFactories/Newsletter.php

```diff
@@ -101,11 +115,20 @@ public function withPostNotificationsType() {
   /**
    * @return Newsletter
    */
-  public function withWelcomeType() {
+  public function withPostNotificationHistoryType() {
+    $this->data['type'] = \MailPoet\Models\Newsletter::TYPE_NOTIFICATION_HISTORY;
+    $this->withOptions([]);
+    return $this;
+  }
+
+  /**
+   * @return Newsletter
+   */
+  public function withWelcomeType($segment_id = 2) {
```

Comment:
Thanks for pointing this to me. I was not aware of different welcome emails. This factory method had originally hardcoded value for `welcome` email type with event type `segment` so I'll just change the name of the method so that it refers to the event type.